parsely_raw_data/__init__.py

```diff
@@ -28,3 +29,39 @@
     'stream',
     'utils',
 ]
+
+BOOLEAN_FIELDS = [
+    "flags_is_amp",
+]
+
+
+def normalize_keys(r, schema):
+    """Conform events to public schema: correct keys and proper value types."""
+    schema = schema or schema.SCHEMA
+    event_dict = {}
+    version =__version__
+
+    # fix value types
+    if r.get("metadata.share_urls") is not None and isinstance(
+        r["metadata.share_urls"], dict
+    ):
+        r["metadata.share_urls"] = list(r["metadata.share_urls"].values()) or None
+
+    # emit only public schema items
+    for key, val in iteritems(r):
+        key = key.replace(".", "_")
+        if key in schema:
+            event_dict[key] = val
+
+    # ensure all columns are available and null when needed
+    # account for all boolean schema defined fields as this is parsely_raw_data specific
+    for key in schema:
+        if key not in event_dict.keys():
+            if key in BOOLEAN_FIELDS:
+                event_dict[key] = False
+            else:
+                event_dict[key] = None
```

Comment:
I wonder if you could combine these two loops. That is, iterate over `schema` once, try to pull the value from `r`, and if it's not there put in a default value.  Functionally the same, but means iterating over everything once instead of twice. I think it's also be more readable.

Reply:
the first loop loops over the event_dict to make sure that the keys align with the schema.

the second loop loops over the schema to add any missing keys.

i see you point about combining them, but it's not clear how to remove keys from `r` without looping over them too, since the set of `r` keys might be > than the set of `schema` keys.